gravitee-node-services/gravitee-node-services-upgrader/src/test/java/io/gravitee/node/services/upgrader/UpgraderServiceImplTest.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.node.services.upgrader;
+
+import static org.mockito.Mockito.*;
+
+import io.gravitee.node.api.Node;
+import io.gravitee.node.api.UpgraderRepository;
+import io.gravitee.node.api.upgrader.Upgrader;
+import io.gravitee.node.api.upgrader.UpgraderData;
+import io.reactivex.Maybe;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.context.ApplicationContext;
+
+/**
+ * @author Kamiel Ahmadpour (kamiel.ahmadpour at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class UpgraderServiceImplTest {
+
+  @Mock
+  private UpgraderRepository repository;
+
+  @Mock
+  private ApplicationContext applicationContext;
+
+  @Mock
+  private Node node;
+
+  @InjectMocks
+  private UpgraderServiceImpl cut;
+
+  @Test
+  public void shouldUpgrade() throws Exception {
```

Comment:
Can you add more tests (ex: multiple upgraders, failure for one of the upgrader, ...) ?

Reply:
I added more tests